### PR TITLE
add the first article about BSP

### PR DESCRIPTION
**Pull request by georgik**
Please, review the article about BSP (@espzav) .
The idea is to have one article for BSP users, then there should be another  one for BSP creators.

### CODE REVIEW

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (igrr) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:18]: I am not sure I would emphasize specifically this part. It is not a goal of ESP-BSP to offer a standardized interface between the boards, or to make porting applications between boards easier. The main goal of ESP-BSP is to offer a convenient way to start a project for a particular development board, without having to look for drivers and other details for that specific board.

Due to the fact that now we have common examples between BSPs, the APIs of different BSPs also have a lot in common. But this is more of a side effect rather than the project goal.

I would suggest rewriting this paragraph focusing on the ease of getting started, and not promising standardized APIs or portability between boards.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (igrr) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:37]: Since we are using specific version of the package below, I would add a note here, which IDF version was used when writing this tutorial.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Thanks. Text updated.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Yep, make sense. I will fix it to 5.3.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (pedrominatel) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:50]: Should we mention that the BSP can be added to a project directly?
i.e:

```bash
idf.py add-dependency "espressif/esp_bsp_generic^1.2.0"
```

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (pedrominatel) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:73]: Consider recording the output with asciinema.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (pedrominatel) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:119]: Maybe adding the code should help visualize the example if the code it's not that big.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:50]: That actually brings an interesting idea. If we would have remove-dependency, we would have a seqequence of commands for switching BSP from CLI.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:76]: Done. @f-hollow The media file is with the article. IMO it's better when article and it's media is collocated, there is already featured.webp, so it kind of make same to have this content there.

I'm ok with moving asciinema to different location, but please explain why the other location should be better.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:122]: Which code do you mean? The app, it's stored as part of the example at BSP site.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:53]: There is add-dependency for the correction of I2C.
I would explain construction of example from the scratch in a different article. This is quick bootstrapping from existing example code.

**Comment by georgik** [comment]
Ready to be merged

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (igrr) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:134]: Sorry, I know you like Wokwi a lot, but it seems to be not really related to the article? Of course we can simulate applications based on ESP-BSP in Wokwi, but so can we do for any other applications. I guess we won't be linking Wokwi to every single tutorial we publish?

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (igrr) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:39]: IDF v5.3 has not been released yet, and the `stable` link doesn't necessarily link to IDF v5.3. It points to v5.2.2 now, it will point to v5.3 at some point, and it might point to v5.4 in the future.

I suggest updating the link to point a fixed release and updating the link text to match that release.

**Comment by igrr** [comment]
LGTM aside from the couple remaining comments!

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (pedrominatel) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:122]: You are using the `display_audio_photo` example so that we can add some code, especially the BSP initialization.

```c
    /* Initialize and mount SPIFFS */
    bsp_spiffs_mount();

    /* Initialize I2C (for touch and audio) */
    bsp_i2c_init();

    /* Initialize display and LVGL */
    bsp_display_start();

    /* Set default display brightness */
    bsp_display_brightness_set(APP_DISP_DEFAULT_BRIGHTNESS);

    /* Add and show LVGL objects on display */
    app_disp_lvgl_show();

    /* Initialize SPI flash file system and show list of files on display */
    app_disp_fs_init();

    /* Initialize audio */
    app_audio_init();

    ESP_LOGI(TAG, "Example initialization done.");
```

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (pedrominatel) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:134]: or replace with [Try with ESP Launchpad](https://espressif.github.io/esp-launchpad/?flashConfigURL=https://espressif.github.io/esp-bsp/config.toml&app=display_audio_photo)

Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:139]: Launchpad is great idea, and actually BSPs has ready everything even binaries. I've added the button, it works out of box.

Just the issue is that there is Zoom mouse pointer instead of Hand pointer when hover over the image. @f-hollow Do you know how to resolve this issue with mouse pointer?

**Comment by georgik** [comment]
We just discovered, that ESP-BSP will create the example with the configuration stored in repo, so idf_component.yml has dependency on esp-box even when generated for ESP32-S3-BOX-3. It's not solved yet, and requires manual update of idf_component.yaml. I need to figure out way how to add it to article without making it too complex.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (pedrominatel) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:139]: You can use this shortcode:

```html
{{< figure
    src="assets/image.png"
    href="https://www.espressif.com/"
    alt="Image"
    nozoom=true
    caption="Photo by..."
>}}
```

Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:170]: @pedrominatel Unfortunately this disable the link, it's not clickable.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:158]: That's good idea. I'll add an paragraph

Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:186]: The code is in place. Good idea, makes it easier to understand the concept of BSP API.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Good point. Updated to release v5.3. I expect to update the article in the future, e.g. to remove the patch parts with esp_codec_dev, once BSP gets fixed.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: @igrr I've added a chapter about possibility to simulate the S3-BOX also in context of available IDEs that Espressif or third party provides.

We were discussing with @urish an option how to run the binary directly from the article, the API is not ready yet, in future it might be possible to replace this part directly with simulation of the mentioned example in the article page.

Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Regading Launchpad link, I've tried several options, none of them work.
I've removed the image for the time-being and replaced it by simple link.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:125]: Solved. Assets should be co-located.

**Comment by georgik** [comment]
All corrections are in place, waiting for final review @f-hollow .

**Review by pedrominatel — APPROVED** [review]

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (f-hollow) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:189]: Downloading code files directly from a website to include in your project feels like a workaround for lack of a better choice these days. At least, I went through all kinds of tutorials myself in recent months and do not remember any cases where I had to download code files directly from an article.

Can these files be stored in a repo? Or maybe it would be better to give them in code blocks in the article directly?

**Review by f-hollow — APPROVED** [review]
@georgik Thank you for contributing yet another great article!

Left a few small nitpicks. Sorry for my delayed review!

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (f-hollow) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Me and @georgik had a separate conversation and ended up using the [button shortcode](https://blowfish.page/docs/shortcodes/#button) included in the Blowfish theme.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Good point. We need to find a way how to talk about these boards.
Replaced by: development boards with Espressif SoCs like ESP32 or ESP32-S3

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Good point. Link added.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:45]: Great idea. I've added intro-screen using the following script: https://github.com/georgik/esp32-developer-portal-asciinema-toolbox/blob/main/support/intro-screen.py

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Fixed.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md:189]: True, makes sense. Added into a content, those files are not too big.

**Comment by tore-espressif** [comment]
@georgik Is there an easy way to preview the new article?

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (tore-espressif) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:16]: Doesn't have to be changed:

Should we also mention here that esp-bsp works with esp-idf only? that is, it does not work with arduino?

**Review by tore-espressif — APPROVED** [review]
@georgik Great article, Thank you!

**Comment by f-hollow** [comment]
> @georgik Is there an easy way to preview the new article?

Created issue #43

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (f-hollow) [inline content/blog/simplify-embedded-projects-with-esp-bsp/index.md:229]: @georgik Please consider replacing `ESP32 boards` as per the comment above.

All other changes, LGTM!

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Good point. Reformulated. I made it explicit, that it's for ESP-IDF based projects.

File: content/blog/simplify-embedded-projects-with-esp-bsp/index.md
Comment (georgik) [reply content/blog/simplify-embedded-projects-with-esp-bsp/index.md]: Replaced by Espressif SoC.

**Pull request merged**
